src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -970,13 +980,35 @@ public LexerImpl(TexlLexer lex, string text, StringBuilder sb, Flags flags)
 
                 _numberIsFloat = flags.HasFlag(TexlLexer.Flags.NumberIsFloat);
                 _disableReservedKeywords = flags.HasFlag(TexlLexer.Flags.DisableReservedKeywords);
-
-                _modeStack = new Stack<LexerMode>();
-                _modeStack.Push(LexerMode.Normal);
+                _textFirst = flags.HasFlag(TexlLexer.Flags.TextFirst);
+
+                _modeStack = new Stack<LexerMode>();
+
+                if (_textFirst)
+                {
+                    int ich = 0;
+
+                    // skip whitespace looking for a leading '=' that starts a formula
+                    while (ich < _charCount && char.IsWhiteSpace(text[ich]))
+                    {
+                        ich++;
+                    }
+
+                    if (ich < _charCount && text[ich] == '=')
+                    {
+                        // skip '=' and treat the rest as a normal formula
+                        CurrentPos = ich + 1;
+                    }
+                    else
+                    {
+                        // CurrentPos doesn't move (stays at start of buffer) and treat all of it as text first

```

Comment:
>CurrentPos doesn't move (stays at start of buffer) [](http://example.com/codeflow?start=27&length=50)

Can you add a test where we have whitespaces in the beginning? Something like

    >>      hello
    "     hello"

Reply:
Done.  I couldn't add it in the .txt files as we strip the leading whitespace when running the tests, but I added it in C# tests.